server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -1556,14 +1556,18 @@ public long fileLength(String name) throws IOException {
 
     public void testRefreshMetric() throws IOException {
         IndexShard shard = newStartedShard();
-        assertThat(shard.refreshStats().getTotal(), equalTo(2L)); // refresh on: finalize and end of recovery
+        // refresh on: finalize and end of recovery
+        // finalizing a replica involves two refreshes with soft deletes because of estimateNumberOfHistoryOperations()
+        final long initialRefreshes = shard.routingEntry().primary() || shard.indexSettings().isSoftDeleteEnabled() == false ? 2L : 3L;
```

Comment:
I'm not sure if this is expected or not. Seems a bit awkward. Maybe there's a better solution?